dali/kernels/slice/slice_gpu.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SLICE_SLICE_GPU_H_
+#define DALI_KERNELS_SLICE_SLICE_GPU_H_
+
+#include <cuda_runtime.h>
+#include <vector>
+#include "dali/core/common.h"
+#include "dali/core/dev_array.h"
+#include "dali/error_handling.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+
+template <std::size_t Dims>
+struct SliceArgsDev {
+  DeviceArray<int64_t, Dims> in_strides;
+  DeviceArray<int64_t, Dims> out_strides;
+  DeviceArray<int64_t, Dims> anchor;
+};
+
+template <typename OutputType, typename InputType, std::size_t Dims>
+__global__ void SliceKernel(OutputType *__restrict__ output,
+                            const InputType *__restrict__ input,
+                            SliceArgsDev<Dims> slice_args_dev,
```

Comment:
I'd consider batch processing here - but maybe that's a topic for a followup?

Reply:
Follow up task created: DALI-805